helpers/tests/unit/test_log_context.py

```diff
@@ -0,0 +1,112 @@
+import asyncio
+
+from asgiref.sync import async_to_sync
+from sqlalchemy.exc import IntegrityError
+
+from database.tests.factories.core import CommitFactory, OwnerFactory, RepositoryFactory
+from helpers.log_context import (
+    LogContext,
+    get_log_context,
+    set_log_context,
+    update_log_context,
+)
+
+
+def create_db_records(dbsession):
+    owner = OwnerFactory.create(
+        service="github",
+        username="codecove2e",
+        unencrypted_oauth_token="test76zow6xgh7modd88noxr245j2z25t4ustoff",
+    )
+    dbsession.add(owner)
+
+    repo = RepositoryFactory.create(
+        owner=owner,
+        yaml={"codecov": {"max_report_age": "1y ago"}},
+        name="example-python",
+    )
+    dbsession.add(repo)
+
+    commit = CommitFactory.create(
+        message="",
+        commitid="c5b67303452bbff57cc1f49984339cde39eb1db5",
+        repository=repo,
+    )
+    dbsession.add(commit)
+
+    dbsession.commit()
+    dbsession.expire(owner)
+    dbsession.expire(repo)
+    dbsession.expire(commit)
+
+    return owner, repo, commit
+
+
+def test_populate_just_owner(dbsession):
+    owner, _repo, _commit = create_db_records(dbsession)
+    log_context = LogContext(owner_id=owner.ownerid)
+    log_context.populate_from_sqlalchemy(dbsession)
+
+    assert log_context == LogContext(owner_id=owner.ownerid)
+
+
+def test_populate_just_repo(dbsession):
+    owner, repo, _commit = create_db_records(dbsession)
+    log_context = LogContext(repo_id=repo.repoid)
+    log_context.populate_from_sqlalchemy(dbsession)
+
+    assert log_context == LogContext(repo_id=repo.repoid, owner_id=owner.ownerid)
+
+
+def test_populate_just_commit_sha(dbsession):
+    _owner, _repo, commit = create_db_records(dbsession)
+    log_context = LogContext(commit_sha=commit.commitid)
+    log_context.populate_from_sqlalchemy(dbsession)
+
+    assert log_context == LogContext(commit_sha=commit.commitid)
```

Comment:
Couldn't you in theory know the repo/owner from the commit sha? Or are you assuming sha's here shouldn't be unique, hence you could find multiple records with the same sha? If so, what about a commit sha/id combo?

Reply:
mirrors/forks will have their own copies of the same SHAs so just SHA is not enough. if you have the commit's DB ID it's enough to find everything else, but that case was not implemented because it never comes up in our actual code. we don't really pass around the commit's DB ID ever

Reply:
Makes sense, hadn't considered forks